src/service/price/PriceService.ts

```diff
@@ -1,7 +1,154 @@
+		return tokenPrice.usd * balance;
+	}
+
+	/**
+	 * Retrieve the ETH price for a given token balance.
+	 * @param contract Token contract address.
+	 * @param balance Token balance to calculate price.
+	 */
+	async getEthValue(contract: string, balance: number): Promise<number> {
+		const tokenPrice = await this.getTokenPriceData(contract);
+		return tokenPrice.eth * balance;
+	}
+
+	/**
+	 * Retrieve the price data for a given token in USD and ETH.
+	 * @param contract Token contract address.
+	 */
+	async getTokenPriceData(contract: string): Promise<TokenPrice> {
+		const cachedPrice: TokenPrice | undefined = priceCache.get(contract);
+		if (!cachedPrice) {
+			const priceData = await this.getPriceData();
+			return priceData[contract];
+		}
+		return cachedPrice;
+	}
+
+	/**
+	 * Retrieve all protocol token prices in both USD and ETH.
+	 */
+	async getPriceData(): Promise<PriceData> {
+		const [
+			badgerPrice,
+			diggPrice,
+			wbtcPrice,
+			uniBadgerWbtcPrice,
+			uniDiggWbtcPrice,
+			slpBadgerWbtcPrice,
+			slpDiggWbtcPrice,
+			slpEthWbtcPrice,
+			renBtcPrice,
+			sBtcPrice,
+			tBtcPrice,
+			wethPrice,
+			sushiPrice,
+		] = await Promise.all([
+			this.fetchContractPrice(TOKENS.BADGER),
+			this.fetchContractPrice(TOKENS.DIGG),
+			this.fetchContractPrice(TOKENS.WBTC),
+			getUniswapPrice(TOKENS.UNI_BADGER_WBTC),
+			getUniswapPrice(TOKENS.UNI_DIGG_WBTC),
+			getSushiswapPrice(TOKENS.SUSHI_BADGER_WBTC),
+			getSushiswapPrice(TOKENS.SUSHI_DIGG_WBTC),
+			getSushiswapPrice(TOKENS.SUSHI_ETH_WBTC),
+			this.fetchContractPrice(TOKENS.CRV_RENBTC),
+			this.fetchTokenPrice('sbtc'),
+			this.fetchTokenPrice('tbtc'),
+			this.fetchContractPrice(TOKENS.WETH),
+			this.fetchContractPrice(TOKENS.SUSHI),
+		]);
+		const priceData: PriceData = {};
+		priceData[TOKENS.BADGER] = badgerPrice;
+		priceData[TOKENS.DIGG] = diggPrice;
+		priceData[TOKENS.WBTC] = wbtcPrice;
+		priceData[TOKENS.UNI_BADGER_WBTC] = uniBadgerWbtcPrice;
+		priceData[TOKENS.UNI_DIGG_WBTC] = uniDiggWbtcPrice;
+		priceData[TOKENS.SUSHI_BADGER_WBTC] = slpBadgerWbtcPrice;
+		priceData[TOKENS.SUSHI_DIGG_WBTC] = slpDiggWbtcPrice;
+		priceData[TOKENS.SUSHI_ETH_WBTC] = slpEthWbtcPrice;
+		priceData[TOKENS.CRV_RENBTC] = renBtcPrice;
+		priceData[TOKENS.CRV_SBTC] = sBtcPrice;
+		priceData[TOKENS.CRV_TBTC] = tBtcPrice;
+		priceData[TOKENS.WETH] = wethPrice;
+		priceData[TOKENS.SUSHI] = sushiPrice;
+		return priceData;
+	}
+
+	/**
+	 * Fetch a token contract price from CoinGecko.
+	 * @param contract Token contract address.
+	 */
+	async fetchContractPrice(contract: string): Promise<TokenPrice> {
+		return getContractPrice(contract);
+	}
+
+	/**
+	 * Fetch a token price from CoinGecko. Token name must match the name used by CoinGecko.
+	 * e.g. https://www.coingecko.com/en/coins/badger-dao => badger-dao
+	 * @param token Token name.
+	 */
+	async fetchTokenPrice(token: string): Promise<TokenPrice> {
+		return getTokenPrice(token);
+	}
```

Comment:
Same as above... also why the switch from get to fetch (or fetch to get depending on how you look at things).